### PR TITLE
Feature: Align elements

### PR DESCRIPTION
**Pull request by SvenKube**
This pull request implements the alignment feature suggested in #772

In order to allow for a sensible integration with the existing context menu items it also adds a `contextMenuOrder` property to z-index actions.

### CODE REVIEW

File: src/components/ShortcutsDialog.tsx [hunk 1]
```diff
@@ -280,6 +280,22 @@ export const ShortcutsDialog = ({ onClose }: { onClose?: () => void }) => {
...
+                shortcuts={[getShortcutKey("CtrlOrCmd+Shift+🠟")]}
+              />
+              <Shortcut
+                label={t("labels.alignLeft")}
+                shortcuts={[getShortcutKey("CtrlOrCmd+Shift+🠜")]}
+              />
+              <Shortcut
+                label={t("labels.alignRight")}
+                shortcuts={[getShortcutKey("CtrlOrCmd+Shift+🠞")]}
+              />
```
Comment (dwelle) [inline src/components/ShortcutsDialog.tsx:298]: I'd use regular words here (Left, Right...).

**Comment by dwelle** [comment]
Nice, thanks for the PR!

Few notes

- The aligning doesn't work well with rotated elements. I can't seem to be able to upload images to GH right now, so can't illustrate. Try rotating to rectangles and top-aligning them. It should align by their bounding boxes.
- Any reason why some of the icons have dotted lines while the others don't?

File: src/actions/actionAlign.tsx [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  alignment: Alignment,
+) {
+  const selectedElements = getSelectedElements(
+    getNonDeletedElements(elements),
+    appState,
+  );
+
+  const updatedElements = alignElements(selectedElements, alignment);
+
+  const updatedElementsMap = new Map(updatedElements.map((e) => [e.id, e]));
```
Comment (dwelle) [inline src/actions/actionAlign.tsx:38]: Let's use existing `getElementMap()` (`src/element/index.ts`) helper here.

**Comment by IllTemperedMax** [comment]
Thanks for the quick feedback. I guess we'll have to give aligning rotated shapes some more thought.

> * Any reason why some of the icons have dotted lines while the others don't?

The idea behind this is that for the centering actions the reference line crosses through the elements which it doesn't for the other four. I actually tried having a solid line both in front of and behind the blocks. Neither we're as clear as the one with the dashed line I decided on at such small scale in the application.

File: src/components/ShortcutsDialog.tsx (hunk 1 above)
Comment (SvenKube) [reply src/components/ShortcutsDialog.tsx]: Yeah, good idea. Changed that already.

**Comment by lipis** [comment]
Besides the rotated elements is awesome!

**Review by lipis — CHANGES_REQUESTED** [review]
Fix rotation and remove items from the content menu.

**Comment by dwelle** [comment]
Working great.

Final thing, are we fine with the button order? Traditionally, vertical and horizontal buttons are grouped together. Ours:

image

I can see benefits of our solution, but it's non-standard, and it makes it harder to realize (if you're not already familiar with the behavior) that you can align by two different axes at the same time.

**Comment by IllTemperedMax** [comment]
I changed the order of the actions in the respective component but anywhere else in the code.

For some reason the tests suddenly fail in a spot we didn't touch and I can't see any connection either. Is that just a random hickup that was there before? Also, it appears I can't rerun the check without making another push...

**Comment by dwelle** [comment]
> For some reason the tests suddenly fail in a spot we didn't touch and I can't see any connection either. Is that just a random hickup that was there before?

Checking the run, yes, it's failing on `__cokieJar` which is an unrelated flake fixed in my recent PR. Will merge it soon and then merge it into this one.

**Review by lipis — APPROVED** [review]

**Comment by dwelle** [comment]
Shipping! Superb job @IllTemperedMax @SvenKube :tada:

**Pull request merged**